lib/Sema/CSApply.cpp

```diff
@@ -4732,9 +4732,19 @@ namespace {
               KeyPathExpr::Component::forOptionalChain(objectTy, loc));
           break;
         }
-        case KeyPathExpr::Component::Kind::OptionalForce:
-          buildKeyPathOptionalForceComponent(resolvedComponents);
+        case KeyPathExpr::Component::Kind::OptionalForce: {
+          // Handle force optional when it is the first component e.g.
+          // \String?.!.count
+          if (resolvedComponents.empty()) {
+            auto loc = origComponent.getLoc();
+            resolvedComponents.push_back(
+                KeyPathExpr::Component::forOptionalForce(
+                    componentTy->lookThroughSingleOptionalType(), loc));
```

Comment:
Could we assert that `componentTy` is an optional, just to make sure this isn't a silent no-op?

Reply:
Makes sense to me and assert would be good here, also you could use `getOptionalObjectType()` instead of `lookThroughSingleOptionalType`.

Reply:
Sure! Just done